Templates/AppSource App/.github/workflows/TroubleShooting.yaml

```diff
@@ -0,0 +1,60 @@
+name: 'Troubleshooting'
+
+on:
+  workflow_dispatch:
+    inputs:
+      displayNameOfSecrets:
+        description: Display the name (not the value) of secrets available to the repository
+        type: boolean
+        default: true
+      useLatestTroubleshooting:
+        description: Use the latest troubleshooting action from https://github.com/microsoft/AL-Go
+        type: boolean
+        default: false
+
+permissions:
+  contents: write
+  actions: read
+
+defaults:
+  run:
+    shell: powershell
+
+env:
+  ALGoOrgSettings: ${{ vars.ALGoOrgSettings }}
+  ALGoRepoSettings: ${{ vars.ALGoRepoSettings }}
+
+jobs:
+  Troubleshooting:
+    name: microsoft/AL-Go-Actions/TroubleShooting@main
+    if: github.event.inputs.useLatestTroubleshooting == 'false'
+    runs-on: [ windows-latest ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          lfs: true
+
+      - name: Troubleshooting
+        uses: microsoft/AL-Go-Actions/TroubleShooting@main
+        with:
+          shell: powershell
+          gitHubSecrets: ${{ toJson(secrets) }}
+          displayNameOfSecrets: ${{ github.event.inputs.displayNameOfSecrets }}
+
+  LatestTroubleshooting: 
+    name: microsoft/AL-Go/Actions/TroubleShooting@main
```

Comment:
The thing that really comes to mind is that if somebody for some reason cannot run Update AL-Go System Files. This would happen if a settings file is corrupted and we do check that - but if something else suddenly happened - let's say 2 years from now, some GitHub action suddenly was no longer supported and version 4.1 of AL-Go stopped working - then people coming back to this repo 3 years from now cannot run Update AL-Go system files and troubleshooting wouldn't tell them anything.

Reply:
That was my concern :) In some cases that means loading AL-Go-Helper.ps1 from microsoft/AL-Go/Actions. As Maria also mentioned, this is unreleased code that hasn't gone through the E2E tests yet

Reply:
> Or do you mean that it should use microsoft/AL-Go-Actions@main instead?

I think that is what she said 😄 microsoft/**AL-Go-Actions**@main - Not microsoft/**AL-Go/Actions**@main

> Problem with this is that nothing uses that. Everything gets changed to use @v4.1

Maybe nothing uses that today but is there a problem with using it? Doesn't microsoft/AL-Go-Actions@main always contain the latest **released** version of AL-Go? microsoft/AL-Go/Actions@main contains our latest **unreleased** code. 

IMO it'd be more stable to uses the Troubleshooting action out of microsoft/AL-Go-Actions@main since anything we deploy there has gone through E2E testing.